### PR TITLE
ISPN-9329 NPE during cache start due to circular component dependencies

### PR DESCRIPTION
**Pull request by wburns**
* Don't wrap custom IDC
* Move circular dependency into GroupManagerImpl instead

https://issues.jboss.org/browse/ISPN-9329

### CODE REVIEW

File: core/src/main/java/org/infinispan/distribution/group/impl/GroupManagerImpl.java [hunk 1]
```diff
@@ -62,7 +64,8 @@ else if (possibleMethods.size() == 1)
             throw new IllegalStateException(Util.formatString("Cannot define more that one @Group method for class hierarchy rooted at %s", clazz.getName()));
     }
 
-    @Inject private DistributionManager distributionManager;
+    @Inject private ComponentRegistry componentRegistry;
+    private DistributionManager distributionManager;
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/distribution/group/impl/GroupManagerImpl.java]: Please move the comment about the dependency cycle here

Comment (wburns) [reply core/src/main/java/org/infinispan/distribution/group/impl/GroupManagerImpl.java]: Sure

**Comment by wburns** [comment]
Updated

**Comment by wburns** [comment]
Imports should be fixed now as well.

**Comment by pferraro** [comment]
@wburns FYI - while you're touching this class, it looks like DefaultSegmentedDataContainer contains a unnecessary @Listener annotation and calls cache.removeListener(...) during @Stop, even though it has no corresponding listener registration nor does it handle any events.

**Comment by wburns** [comment]
It is harmless. I actually already fixed it in a different upcoming PR https://github.com/wburns/infinispan/commit/9a5de439afcc58f08008222b3d43c8b6034489ab#diff-6d6afaa6691141286cacb1e0057e6dcf

**Comment by pferraro** [comment]
I realize it's harmless - it was just a little confusing. :)

**Comment by danberindei** [comment]
Integrated, thanks Will!

**Pull request merged**